cns/restserver/ipam.go

```diff
@@ -112,6 +113,24 @@ func (service *HTTPRestService) MarkIPsAsPending(numberToMark int) (map[string]c
 	return nil, fmt.Errorf("Failed to mark %d IP's as pending, only marked %d IP's", numberToMark, len(pendingReleaseIPs))
 }
 
+// UpdatePendingProgrammingIPs will update pending programming IPs to available if
+// NMAgent side's programmed NC version keep up with NC version with secondary IP.
+// This function must be called in a service lock.
+func (service *HTTPRestService) UpdatePendingProgrammingIPs(nmagentNCVersion string, req cns.CreateNetworkContainerRequest) {
```

Comment:
Can you rename it to MarkIpsAsAvailableUntransacted. Reason for untransacted again because caller is taking care of the lock.

Reply:
We also need to log how much time it took for nmagent to actually update the version. Its okay to capture it in a different PR, but please add a WI to track the work.

Reply:
1. Sure, I'll change the name.
2. I'll add it in this PR, only need 2 more extra lines.

Reply:
1. I'll Updated name to MarkIpsAsAvailableUntransacted
2. I'll add time consuming in this PR. It's a 2 line change.